overlord/devicestate/handlers_install.go

```diff
@@ -1278,7 +1295,19 @@ func (m *DeviceManager) doInstallSetupStorageEncryption(t *state.Task, _ *tomb.T
 		return fmt.Errorf("reading gadget information: %v", err)
 	}
 
-	encryptInfo, err := m.encryptionSupportInfo(systemAndSeeds.Model, secboot.TPMProvisionFull, systemAndSeeds.InfosByType[snap.TypeKernel], gadgetInfo)
```

Comment:
actually would it make sense to add the snapd version structure to `systemAndEssentialSnaps` type and populate it in `loadAndMountSystemLabelSnaps()` ?

Reply:
As @pedronis explained we don't expect the relevant code paths to only run for UC20+, I embedded the snapd version structure to `systemAndEssentialSnaps`. Thanks for the suggestion!

Reply:
https://github.com/canonical/snapd/pull/14943#discussion_r1932720748